### PR TITLE
Build wheels for Python 3.10 also

### PR DESCRIPTION
**Pull request by QuLogic**
This is based on #62 because nose is broken on 3.10.

### CODE REVIEW

**Comment by djhoese** [comment]
Doesn't the `python setup.py test` line in the CI have to be updated to use pytest too?

What's the difference between manylinux and manylinux2010 as far as supporting users go? Can manylinux2010 be used for older versions of python or does that cut off a significant number of users?

File: scripts/build-manylinux-wheels.sh [hunk 1]
```diff
@@ -25,12 +25,25 @@ mkdir -p /io/temp-wheels
...
 
 # Iterate through available pythons.
-for PYBIN in /opt/python/cp3[6789]*/bin; do
-    "${PYBIN}/pip" install -q -U setuptools wheel nose --cache-dir /io/pip-cache
-    # Run the following in root of this repo.
-    (cd /io/ && USE_OMP=$USE_OMP "${PYBIN}/pip" install -q .)
-    (cd /io/ && USE_OMP=$USE_OMP "${PYBIN}/python" setup.py nosetests)
-    (cd /io/ && USE_OMP=$USE_OMP "${PYBIN}/python" setup.py -q bdist_wheel -d /io/temp-wheels)
+for PY in cp3{6,7,8,9,10}; do
+    if [[ $PLAT == manylinux2010* ]]; then
```
Comment (djhoese) [inline scripts/build-manylinux-wheels.sh:29]: Doesn't this require `=~` instead of `==`?

Comment (QuLogic) [reply scripts/build-manylinux-wheels.sh:29]: No, it's a pattern match, not a regex.

**Comment by QuLogic** [comment]
> Doesn't the `python setup.py test` line in the CI have to be updated to use pytest too?

Ah yes, but that doesn't run for me, so I'm not sure if it's right.

> What's the difference between manylinux and manylinux2010 as far as supporting users go? Can manylinux2010 be used for older versions of python or does that cut off a significant number of users?

All I can definitively say is that `manylinux1` images do not provide Python 3.10. But see the table in https://github.com/pypa/manylinux/

File: scripts/build-manylinux-wheels.sh (hunk 1 above)
Comment (djhoese) [reply scripts/build-manylinux-wheels.sh:29]: :man_facepalming: Thanks

**Comment by djhoese** [comment]
>> Doesn't the python setup.py test line in the CI have to be updated to use pytest too?
>
> Ah yes, but that doesn't run for me, so I'm not sure if it's right.

You mean running `pytest pykdtree` does run for you?

> All I can definitively say is that manylinux1 images do not provide Python 3.10. But see the table in https://github.com/pypa/manylinux/

Maybe I'm reading it wrong, but none of the images currently provide 3.10. I see that in the github issue linked on that README that manylinux1 is being deprecated and won't get 3.10 in the future. But the list of versions in that table are just what bundled version of pip will support wheels generated with that. I think this means that moving all pykdtree wheel builds to manylinux2010 will still work for Python 3.6+, but Python 3.6 users will need to upgrade their version of pip to be able to install the wheel.

Anyway...we'll have to wait for @mraspaud as I don't have permissions to do anything on this repository. I still think the CI pytest line should be updated and I vote for manylinux2010 being used everywhere.

**Comment by QuLogic** [comment]
> > > Doesn't the python setup.py test line in the CI have to be updated to use pytest too?
> > 
> > 
> > Ah yes, but that doesn't run for me, so I'm not sure if it's right.
> 
> You mean running `pytest pykdtree` does run for you?

No, I mean AppVeyor doesn't run on PRs.

> > All I can definitively say is that manylinux1 images do not provide Python 3.10. But see the table in https://github.com/pypa/manylinux/
> 
> Maybe I'm reading it wrong, but none of the images currently provide 3.10. I see that in the github issue linked on that README that manylinux1 is being deprecated and won't get 3.10 in the future.

There is 3.10 in manylinux2010, or the workflow wouldn't have passed: https://github.com/QuLogic/pykdtree/actions/runs/1263366622

> But the list of versions in that table are just what bundled version of pip will support wheels generated with that. I think this means that moving all pykdtree wheel builds to manylinux2010 will still work for Python 3.6+, but Python 3.6 users will need to upgrade their version of pip to be able to install the wheel.
> 
> Anyway...we'll have to wait for @mraspaud as I don't have permissions to do anything on this repository. I still think the CI pytest line should be updated and I vote for manylinux2010 being used everywhere.

I fixed the AppVeyor line (I think, but don't know since it doesn't run.)

**Comment by djhoese** [comment]
> There is 3.10 in manylinux2010, or the workflow wouldn't have passed: https://github.com/QuLogic/pykdtree/actions/runs/1263366622

Ah cool. The README needs to be updated then.

**Pull request merged**